include/seqan3/range/view/repeat_n.hpp

```diff
@@ -0,0 +1,92 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::view::char_to.
+ */
+
+#pragma once
+
+#include <seqan3/range/view/repeat.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief The underlying type of seqan3::view::trim.
+ * \ingroup view
+ *
+ * Under the hood this delegates to view::take_until.
+ */
+struct repeat_n_fn
+{
+    /*!\brief todo
+     * \tparam irng_t The type of the range being processed. See seqan3::view::trim for requirements.
+     * \param irange The range being processed.
+     * \param threshold The minimum quality as a phred score [integral type].
+     */
+    template <typename value_t>
+    constexpr auto operator()(value_t && value, size_t const count) const
+    {
+        static_assert(std::CopyConstructible<value_t>, "The value passed to repeat_n must be copy constructible.");
+
+        return view::repeat(std::forward<value_t>(value)) | view::take_exactly(count);
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+/*!\brief A view factory that repeats a given value infinitely.
+ * \tparam    value_t  The type of value to repeat wrapped in a std::view::single; must model std::CopyConstructible.
```

Comment:
The term "must model..." is established for concepts, right? Then please note that `repeat_n` uses `static_assert` instead of a concept requirement.

Reply:
well but I static_assert the concept, so I thought it still `models` it? I would rather not document how we enforce the concept (requires or static_assert) for the user. They only have to know **that it is** enforced. I chose static_assert because of better error messages :) and we decided on a strategy meeting that whenever the concept is not required for overload resolution, a static_assert is more user-friendly. I think I forgot to write this down after the meeting for everyone. Sorry :(